pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java

```diff
@@ -581,6 +586,107 @@ public static String[] split(String input, String delimiter, int limit) {
     return StringUtils.splitByWholeSeparator(input, delimiter, limit);
   }
 
+  /**
+   * @param input an input string for prefix strings generations.
+   * @param maxlength the max length of the prefix strings for the string.
+   * @return generate an array of prefix strings of the string that are shorter than the specified length.
+   */
+  @ScalarFunction
+  public static String[] prefixes(String input, int maxlength) {
+    ObjectList<String> prefixList = new ObjectArrayList<>();
```

Comment:
`ObjectArrayList` is not really buying us anything here.
Given we know the number of prefixes upfront, we can directly allocate the array

Reply:
replaced the objectArrayList with a fixSizeArr. thanks